chpl/chpl-api/src/test/java/gov/healthit/chpl/listener/CacheRefreshTest.java

```diff
@@ -348,11 +348,7 @@ public void testUpdateListingStatusRefreshesCache() throws EntityRetrievalExcept
 
         ListingUpdateRequest updateRequest = new ListingUpdateRequest();
         updateRequest.setListing(listingToUpdate);
-        try {
-            cpController.updateCertifiedProduct(updateRequest);
-        } catch (ValidationException e) {
-            assertEquals(e.getErrorMessages().size(), 3);
```

Comment:
This test was previously expecting to have some errors on the upload call, and then just assert that they were the same as expected, and continue, right? Given that the test doesn't really care about the errors on the update then that made sense. Now, though, won't it fail when it has any errors on the update? If so, then the test will fail for reasons that have nothing to do with what it's testing. Why not, instead of throwing a ValidationException, just catch it and ignore it?

Same comments with below changes related to the update calls

Reply:
I think that's the way most of the tests with update calls are written, the test will fail because of any validation error that occurs during the update call. If something happens inside that method that causes the exception to be thrown then the update will not happen and the test will fail anyway, whether or not we catch and ignore it inside the unit test because of assertions farther down the test. Maybe I'm misunderstanding your comment though.

Reply:
Line 354 expects to find three errors in the update that happened on line 352, right? And then it won't throw any other errors or whatever, and processing will continue with the rest of the method. But if you remove that try/catch block, then when the update happens, any error in the update will be thrown, which causes the test to fail, but we're not testing the update in this test; we're testing the caching refresh thing.

We want to make our tests only fail when the thing we're testing goes wrong. One way to do that would be to put the update in a try block, catch any validationerrors and ignore them, and then continue. Another could be to mock the update call, though if the update call is supposed to do the cache refresh then that wouldn't work.

Does the cache refresh happen as part of the update?

Reply:
That try catch block on 354 was something I added because of the warnings that got turned into error messages. I originally thought that I could just catch those error messages and just move on with the rest of the test. For some reason, the test initially passed on my local machine with the try-catch, but when I pushed to DEV that test failed because that update to the certification status to Retired from Active did not happen because the validation error was thrown inside of the update call before the actual update. So, I then made a change to the test data to include UCD Processes so that the validation errors would not be thrown and therefore would not need to catch those errors, which I think is the right way to do this.

Reply:
Okay. If the tests are passing now I don't want to hold this up any longer. I'm going to resolve this conversation and approve the PR. It'll be up to @kekey1 to resolve her conversation then